cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt

```diff
@@ -233,25 +226,21 @@ private class ServiceImpl(
             )
 
             if (consentManagerUtils.shouldTriggerByGdprSample && isGdprInConfig) {
-
                 val pvParams = PvDataParamReq(
                     env = messageReq.env,
                     body = campaignManager.getGdprPvDataBody(messageReq),
                     campaignType = GDPR
                 )
 
-                savePvData(pvParams)
+                postPvData(pvParams)
                     .executeOnLeft {
                         pError(it)
                         return@executeOnWorkerThread
                     }
-                    .executeOnRight {
-                        it.gdpr?.uuid?.let { u ->
-                            campaignManager.gdprUuid = u
-                            campaignManager.gdprConsentStatus = campaignManager.gdprConsentStatus?.copy(
-                                uuid = u
-                            )
-                        }
+                    .executeOnRight { pvDataResponse ->
+                        campaignManager.gdprConsentStatus = campaignManager.gdprConsentStatus?.copy(
+                            uuid = pvDataResponse.gdpr?.uuid
```

Comment:
that could save null as the `uuid`, and the previous code did not change the value of the `uuid` if the `it.gdpr?.uuid` returned `null`

do you think we are ok with refreshing the value of `uuid` with the `null` in this case?

Reply:
Yes. The only places that _should_ update uuid are either `pvData` or post choice responses. Even if the `uuid` is null.